### PR TITLE
Fixing multi-gpu yolov8 training 

### PR DESCRIPTION
**Pull request by corey-nm**
# Test plan

Ran train command with sparse & quant recipe with multiple gpus.

### CODE REVIEW

File: src/sparseml/pytorch/yolov8/trainers.py [hunk 1]
```diff
@@ -293,7 +293,8 @@ def callback_on_train_epoch_start(self):
         if model_is_quantized:
             if self.scaler is not None:
                 self.scaler._enabled = False
-            self.ema.enabled = False
+            if self.ema is not None:
+                self.ema.enabled = False
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/yolov8/trainers.py:297]: pretty funny

**Review by dbogunowicz — DISMISSED** [review]

File: src/sparseml/pytorch/yolov8/trainers.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/pytorch/yolov8/trainers.py:297]: Yeah apparently ema is only initialized in the main process, and is none in the others!

File: src/sparseml/pytorch/yolov8/trainers.py [hunk 2]
```diff
@@ -268,7 +268,7 @@ def _setup_train(self, rank, world_size):
             self.steps_per_epoch = len(self.train_loader)  # / self.accumulate
 
             self.scaler = self.manager.modify(
-                self.model,
+                self.model.module if hasattr(self.model, "module") else self.model,
```
Comment (rahul-tuli) [inline src/sparseml/pytorch/yolov8/trainers.py:271]: is this beacuse DataParallel wraps the model in module?

**Review by rahul-tuli — APPROVED** [review]

File: src/sparseml/pytorch/yolov8/trainers.py (hunk 2 above)
Comment (bfineran) [reply src/sparseml/pytorch/yolov8/trainers.py:271]: interesting, there was at least a patch at some point to make this the default inside of modify, maybe it was a different codepath or the patch wasn't merged in due to testing that never occurred

**Review by bfineran — APPROVED** [review]

**Comment by dbogunowicz** [comment]
@corey-nm just a reminder to force push this, sorry for the mess

**Review by bfineran — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Pull request merged**